modules/stochastic_tools/include/distributions/Beta.h

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Distribution.h"
+
+/**
+ * A class used to generate a Beta distribution
+ */
+class Beta : public Distribution
+{
+public:
+  static InputParameters validParams();
+
+  Beta(const InputParameters & parameters);
+
+  virtual Real pdf(const Real & x) const override;
+  virtual Real cdf(const Real & x) const override;
+  virtual Real quantile(const Real & p) const override;
+
+  static Real pdf(const Real & x, const Real & alpha, const Real & beta);
+  static Real cdf(const Real & x, const Real & alpha, const Real & beta);
+  static Real quantile(const Real & p, const Real & alpha, const Real & beta);
+
+  /// Beta function: B(a,b) = Gamma(a)Gamma(b)/Gamma(a+b)
+  static Real betaFunction(const Real & a, const Real & b);
+  /**
+   * Lower incomplete beta function.
+   * Non-boost implementation from:
+   * https://codeplea.com/incomplete-beta-function-c
```

Comment:
The guy took the implementation from (the book might be around longer):
https://www.cec.uchile.cl/cinetica/pcordero/MC_libros/NumericalRecipesinC.pdf

@article{teukolsky1992numerical,
  title={Numerical recipes in C},
  author={Teukolsky, Saul A and Flannery, Brian P and Press, WH and Vetterling, WT},
  journal={SMR},
  volume={693},
  number={1},
  pages={59--70},
  year={1992}
}

But I am okay with keeping the link as a reference.

Reply:
Oh okay. I will add a link to that pdf instead. Do you want me the reference it in the documentation?

Reply:
Yeah, I think this is a nice book to have in our bib file (unless others already have it).

Reply:
Okay, I used the link in this doxy and referenced the bibtex in Beta.md